From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Fri, 19 Dec 2025 08:46:25 +0100
Subject: [PATCH] Revert "tcp: add const to tcp_try_rmem_schedule() and
 sk_rmem_schedule() skb"

This reverts commit 75dff0584cce79203ee9968c66c7589150fed591.
---
 include/net/sock.h   | 2 +-
 net/ipv4/tcp_input.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/net/sock.h b/include/net/sock.h
index 57c0df29ee964f689ec996a22f01f9fd14ed51da..6327c35038261ef2b23cd4b914c0b1d9812bb1bb 100644
--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -1559,7 +1559,7 @@ __sk_rmem_schedule(struct sock *sk, int size, bool pfmemalloc)
 }
 
 static inline bool
-sk_rmem_schedule(struct sock *sk, const struct sk_buff *skb, int size)
+sk_rmem_schedule(struct sock *sk, struct sk_buff *skb, int size)
 {
 	return __sk_rmem_schedule(sk, size, skb_pfmemalloc(skb));
 }
diff --git a/net/ipv4/tcp_input.c b/net/ipv4/tcp_input.c
index b4fb2cc3873b2b45ea4db2dc1f84253b55a632dc..413f1a4be52238776c6acc0838c6dd96b5454fe1 100644
--- a/net/ipv4/tcp_input.c
+++ b/net/ipv4/tcp_input.c
@@ -4888,7 +4888,7 @@ static void tcp_ofo_queue(struct sock *sk)
 static bool tcp_prune_ofo_queue(struct sock *sk, const struct sk_buff *in_skb);
 static int tcp_prune_queue(struct sock *sk, const struct sk_buff *in_skb);
 
-static int tcp_try_rmem_schedule(struct sock *sk, const struct sk_buff *skb,
+static int tcp_try_rmem_schedule(struct sock *sk, struct sk_buff *skb,
 				 unsigned int size)
 {
 	if (atomic_read(&sk->sk_rmem_alloc) > sk->sk_rcvbuf ||
